### DISCUSSION TITLE
Problem setting default language to pt-BR.

### DISCUSSION THREAD

**Discussion by rodrigomarcell** [abpframework/abp#11873] [Q&A] [answered]
I modified the app.UseAbpRequestLocalization() in OnApplicationInitialization on the WebModule Projetct to the following code:

```
var supportedCultures = new[]
{
    new CultureInfo("pt-BR"),
};
app.UseAbpRequestLocalization(options =>
{
    options.DefaultRequestCulture = new RequestCulture("pt-BR");
    options.SupportedCultures = supportedCultures;
    options.SupportedUICultures = supportedCultures;
    options.RequestCultureProviders = new List<IRequestCultureProvider>
    {
        new QueryStringRequestCultureProvider(),
        new CookieRequestCultureProvider()
    };
});
```

I'm using abp.io version 5.1 and that code is not working for me. I have no idea why. Actually, i've got a weird behavior in my abp application. When I start it without a tenant, it's obey my culture info (pt-BR). But in the login page when I click to switch tenant, the language automatically falls back to english.

Obs: I realize that it sets the cookie .AspNetCore.Culture to value c=en|uic=en which would be the english language. But if I open the DevTools in the browser, navigate untill the tab Application and change the cookie value in runtime to c=pt-BR|uic=pt-BR, then it goes to the portuguese language, refresh the page, the it works.

Question is, Why when I my culture info works when there's not tenant set, but in the exactly moment I switch the tenant (by the login page) it falls back to english again?

**Comment by berkansasmaz** [comment, marked as answer]
https://github.com/abpframework/abp/issues/2469

> Set `Abp.Localization.DefaultLanguage` setting (as constant: `LocalizationSettingNames.DefaultLanguage`) to pt-BR. See https://docs.abp.io/en/abp/latest/Settings#setting-values-in-the-application-configuration to simply set it in the `appsettings.json`.

In short, when you add the following code to `appsettings.json` in addition to what you have done, the problem should be solved.

```
 "Settings": {
    "Abp.Localization.DefaultLanguage": "pt-BR"
  }
```

**Comment by rodrigomarcell** [comment]
Thanks, that worked for me.
